syntax = "proto3";

package p2p;

// The gRPC service definition for file operations.
service FileService {
  // Sends a request for downloading a file's URL
  rpc DownloadFile(DownloadRequest) returns (DownloadResponse);
  // Add a new rpc for listing files
  rpc ListFiles(ListFilesRequest) returns (ListFilesResponse);
}

// The request message containing the file's name.
message DownloadRequest {
  string file_name = 1;
}

// The response message containing the file's URL.
message DownloadResponse {
  string file_url = 1;
}

// Define request and response messages for ListFiles
message ListFilesRequest {
  // This request could be empty if no additional parameters are needed
}

message ListFilesResponse {
  repeated FileInfo files = 1;
}

message FileInfo {
  string file_name = 1;
  string file_url = 2;
  string peer_name = 3;
}